gridstatus/ercot_api/ercot_api.py

```diff
@@ -351,6 +395,108 @@ def _shadow_prices_column_name_mapper(self):
             "ViolationAmount": "Violation Amount",
         }
 
+    def get_historical_data(
+        self,
+        endpoint,
+        start_date,
+        end_date,
+        verbose=False,
+    ):
+        """Retrieves historical data from the given emil_id from start to end date.
+
+        Arguments:
+            endpoint: a string representing a specific ERCOT API endpoint.
+            start_date: the start date for the historical data
+            end_date: the end date for the historical data
+            verbose: if True, will print out status messages
+
+        Returns:
+            [pandas.DataFrame]: a dataframe of historical data
+        """
+        emil_id = endpoint.split("/")[1]
+        links = self._get_historical_data_links(emil_id, start_date, end_date, verbose)
+
+        if not links:
+            raise NoDataFoundException(
+                f"No historical data links found for {endpoint} with",
+                f"time range {start_date} to {end_date}",
+            )
+
+        dfs = []
+
+        for link in tqdm(
+            links,
+            desc="Fetching historical data",
+            ncols=80,
+            disable=not verbose,
+        ):
+            try:
+                # Data comes back as a compressed zip file.
+                response = self.make_api_call(link, verbose=verbose, parse_json=False)
+
+                # Convert the bytes to a file-like object
+                bytes = pd.io.common.BytesIO(response)
+
+                # Decompress the zip file and read the csv
+                dfs.append(pd.read_csv(bytes, compression="zip"))
+
+                time.sleep(0.2)  # Necessary to avoid rate limiting
```

Comment:
not a big deal, but i like to make this a parameter `get_historical_data`, so caller can tweak themselves to avoid rate limit